python: docker: build failure on Darwin

**Describe the bug**

`python37Packages.docker` fails to build on Darwin with the following error message:

```python
builder for '/nix/store/320aynm8i2bnks3ijcyhv69x6k9bcvbf-python3.7-docker-4.1.0.drv' failed with exit code 1; last 10 log lines:
          self.addCleanup(shutil.rmtree, base)
          for d in ['foo', 'bar']:
              os.makedirs(os.path.join(base, d))
          sock = socket.socket(socket.AF_UNIX)
          self.addCleanup(sock.close)
  >       sock.bind(os.path.join(base, 'test.sock'))
  E       OSError: AF_UNIX path too long
```

**To Reproduce**

1. `nix build --option build-use-sandbox true -I . nixpkgs.python37Packages.docker`

**Metadata**

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.0.0, macOS 10.15.1`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.1`
 - channels(smaret): `"nixpkgs-19.03pre170546.0f43790d2ce"`
 - nixpkgs: `/nix/store/kd1bflipda7jjnfqxjfiym9268akzq2c-master.tar.gz`

cc @jonringer

--- a/pkgs/development/python-modules/docker/default.nix
+++ b/pkgs/development/python-modules/docker/default.nix
@@ -1,8 +1,9 @@
-{ stdenv, buildPythonPackage, fetchPypi, isPy27
+{ lib, stdenv, buildPythonPackage, fetchPypi, isPy27
 , backports_ssl_match_hostname
 , mock
 , paramiko
 , pytest
+, pytestCheckHook
 , requests
 , six
 , websocket_client
@@ -17,25 +18,25 @@ buildPythonPackage rec {
     sha256 = "1hdgics03fz2fbhalzys7a7kjj54jnl5a37h6lzdgym41gkwa1kf";
   };
 
+  nativeBuildInputs = [
+    pytestCheckHook
+  ] ++ lib.optional isPy27 mock;
+
   propagatedBuildInputs = [
     paramiko
     requests
     six
     websocket_client
-  ] ++ stdenv.lib.optional isPy27 backports_ssl_match_hostname;
+  ] ++ lib.optional isPy27 backports_ssl_match_hostname;
 
-  checkInputs = [
-    mock
-    pytest
-  ];
-
-  # Other tests touch network
+  pytestFlagsArray = [ "tests/unit" ];
   # Deselect socket tests on Darwin because it hits the path length limit for a Unix domain socket
-  checkPhase = ''
-    ${pytest}/bin/pytest tests/unit/ ${stdenv.lib.optionalString stdenv.isDarwin "--deselect=tests/unit/api_test.py::TCPSocketStreamTest"}
-  '';
+  disabledTests = lib.optionals stdenv.isDarwin [ "stream_response" "socket_file" ];
 
-  meta = with stdenv.lib; {
+  # skip setuptoolsCheckPhase
+  doCheck = false;
+
+  meta = with lib; {
     description = "An API client for docker written in Python";
     homepage = "https://github.com/docker/docker-py";
     license = licenses.asl20;

